rapid7_vulndb/komand_rapid7_vulndb/util/utils.py

```diff
@@ -0,0 +1,311 @@
+        cls.validate(fn, dicts=[query], strings=[plugin_inp_db])
+        typed_query = copy.deepcopy(query)
+        db_type = cls.resolve_db_type(plugin_inp_db)
+        typed_query["type"] = db_type
+        return typed_query
+
+    @classmethod
+    def get_query(cls, query: Dict) -> Dict:
+        fn = sys._getframe().f_code.co_name
+        cls.validate(fn, dicts=[query])
+        try:
+            response = requests.get(cls.search_url, params=query,
+                                    allow_redirects=False)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            raise PluginException(
+                cause="Unexpected http error",
+                data=str(e),
+                assistance="If the issue persists please contact support.",
+            )
+        try:
+            data = response.json()
+        except ValueError as e:
+            raise PluginException(
+                cause="Unexpected data format received from Rapid7 API",
+                data=str(e),
+                assistance="If the issue persists please contact support.")
+        return data
+
+    @classmethod
+    def get_content(cls, identifier: str) -> Dict:
+        fn = sys._getframe().f_code.co_name
+        cls.validate(fn, strings=[identifier])
+        try:
+            response = requests.get(
+                urljoin(cls.content_url, identifier),
+                allow_redirects=False)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            raise PluginException(
+                cause="Unexpected http error",
+                data=str(e),
+                assistance="If the issue persists please contact support.",
+            )
+        try:
+            data = response.json()
+        except ValueError as e:
+            raise PluginException(
+                cause="Unexpected data format received from Rapid7 API",
+                data=str(e),
+                assistance="If the issue persists please contact support.")
+        return cls.make_content_result(data)
+
+    @classmethod
+    def gen_dict(cls, orig: Dict, keys: List[str],
+                 *modifiers: Callable[[Dict], None]) -> Dict:
+        """
+        Generic method that allows to generate new dictionary
+        by extracting subset of the keys form the original dictionary
+        and applying any number of callbacks to modify the final result
+
+        :param orig: original dictionary
+        :param keys: list of keys to extract
+        :param modifiers: callback functions to apply on the generated list
+        :return: generated dictionary
+        """
+        fn = sys._getframe().f_code.co_name
+        cls.validate(fn, dicts=[orig], lists=[keys], callables=modifiers)
+        res = {k: orig[k] for k in keys}
+        for modify in modifiers:
+            modify(res)
+        return res
+
+    @classmethod
+    def validate(cls, origin: str,
```

Comment:
This level of validation might be nice to have available in the plugin SDK but I don't think we will want to implement it for each plugin.